src/link/MachO.zig

```diff
@@ -703,42 +711,65 @@ fn linkWithLLD(self: *MachO, comp: *Compilation) !void {
             Compilation.dump_argv(argv.items);
         }
 
-        const new_argv = try arena.allocSentinel(?[*:0]const u8, argv.items.len, null);
-        for (argv.items) |arg, i| {
-            new_argv[i] = try arena.dupeZ(u8, arg);
-        }
+        // TODO https://github.com/ziglang/zig/issues/6971
+        // Note that there is no need to check if running natively since we do that already
+        // when setting `system_linker_hack` in Compilation struct.
+        if (self.base.options.system_linker_hack) {
+            const result = try std.ChildProcess.exec(.{ .allocator = self.base.allocator, .argv = argv.items });
+            defer {
+                self.base.allocator.free(result.stdout);
+                self.base.allocator.free(result.stderr);
+            }
+            if (result.stdout.len != 0) {
+                std.log.warn("unexpected LD stdout: {}", .{result.stdout});
+            }
+            if (result.stderr.len != 0) {
+                std.log.warn("unexpected LD stderr: {}", .{result.stderr});
+            }
+            if (result.term.Exited != 0) {
```

Comment:
Maybe this should use `if (result.term != .Exited or result.term.Exited != 0)` instead, although I doubt stderr will be useful in those cases anyway.  
Still probably a better error than `access of inactive union field Exited`

Reply:
That's a good point, thanks!